CONTRIBUTING.md

```diff
@@ -113,6 +113,34 @@ exist to show coverage directly in your editor. E.g.:
 - <https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters>
 - <https://github.com/alepez/vim-llvmcov>
 
+### Documentation
+
+Here are some guidelines for writing documentation in Ratatui.  
+Every public API should be documented.
+
+#### Content
+
+The main doc comment should talk about the general features that the widget supports and introduce the concepts pointing to the various methods. Focus on interaction with various features and giving enough information that helps understand why you might want something.  
+Examples should help users understand a particular usage, not test a feature. They should be as simple as possible.
+
+#### Format
+
+- First line is summary, second is blank, third onward is more detail
```

Comment:
I think in order to make it easily understandable at a quick glance, an example of the doc comments above the prototype of a method would be nice, what do you think ?

Reply:
Yes sure!